OpenSim/Analyses/StaticOptimization.cpp

```diff
@@ -553,7 +562,7 @@ int StaticOptimization::begin(const SimTK::State& s )
         sWorkingCopy.setTime(s.getTime());
         sWorkingCopy.setQ(s.getQ());
         sWorkingCopy.setU(s.getU());
-        sWorkingCopy.setZ(s.getZ());
+        // No need to copy Z as Actuators could be disabled/removed from _modelWorkingCopy
```

Comment:
Does this mean that there will be different results across the entire Model if just one Actuator is disabled? Why not copy over the parts of `Z` that are still part of the model?

Reply:
Thanks for the feedback @tkuchida If you actually look at the record method at the heart of SO you'll see that Z is never used/copied between states so it seems unnecessary here as well. Considering that Z contains only muscle states which are never used by SO. This felt totally reasonable. The fact that test cases pass suggests that these Zs were not actually used. On a side note I'm not aware of a method to generically copy Z  state variable values unless we cycle through components, know which ones have them and copy :pois

Reply:
As to whether one disabled actuator changes results, I think yes and I also think that is the correct behavior. Having disabled actuators with min control and bounds in the model where the disabled flag is handled inconsistently would result in unpredictable results as shown in the reported issue.

Reply:
Thanks, @aymanhab. Presumably, someone could create their own Actuator that has some important entry in Z, in which case Static won't work---but that probably isn't the only thing that wouldn't work. If the rationale for removing this line is as you said, I suggest changing the comment to something like "Z not copied because it isn't used by record()".

Reply:
> As to whether one disabled actuator changes results, I think yes and I also think that is the correct behavior.

What I meant was that the *mere* act of disabling a muscle (e.g., disabling a muscle that is attached to Ground at both ends) shouldn't change the results.

Reply:
@tkuchida There's no change in computation whatsoever in the properly handled case with no disabled actuators. The behavior with disabled actuators was incorrect so no guarantee of no-change to this incorrect behavior 😃 The documentation should say something like: results were incorrect when some actuators were disabled and are now totally ignored"

Reply:
> The documentation should say something like: results were incorrect when some actuators were disabled and are now totally ignored"

Ok, that should probably be mentioned in the changelog.